src/adapters/ActivityAdapter.js

```diff
@@ -0,0 +1,20 @@
+import {throwError} from 'rxjs';
+
+/**
+ * This is a base class that defines the interface that maps activity data.
+ * Developers that want to extend `ActivityAdapter` must implement all of its methods,
+ * adhering to the exact parameters and structure of the returned objects.
+ */
+export default class ActivityAdapter {
+  /**
+   * Returns an observable that emits message data of the given ID.
+   *
+   * @param {String} id - ID of the message to get
+   * @returns {Observable<Activity>}
```

Comment:
Are we going to define the shape of the `<Activity>` object in the jsdoc as well?

Reply:
Yes, that's a good idea!